line_profiler/cli_utils.py

```diff
@@ -0,0 +1,149 @@
+    `action='store_true'` options.
+    If `hide_complementary_options` is
+    true, the auto-generated option (all the long flags prefixed
+    with 'no-', e.g. '--foo' is negated by '--no-foo') is hidden
+    from the help text.
+
+    Arguments:
+        parser_like (Any):
+            Object having a method `add_argument()`, which has the same
+            semantics and call signature as
+            `ArgumentParser.add_argument()`.
+        hide_complementary_options (bool):
+            Whether to hide the auto-generated complementary options to
+            `action='store_true'` options from the help text for
+            brevity.
+        *args, **kwargs
+            Passed to `parser_like.add_argument()`
+
+    Returns:
+        action_like (Any):
+            Return value of `parser_like.add_argument()`
+    """
+    if kwargs.get('action') not in ('store_true', 'store_false'):
+        return parser_like.add_argument(*args, **kwargs)
+    kwargs['const'] = kwargs['action'] == 'store_true'
+    kwargs['action'] = 'store_const'
+    kwargs.setdefault('default', None)
+    if kwargs['action'] == 'store_false':
+        return parser_like.add_argument(*args, **kwargs)
+    # Automatically generate a complementary option for a boolean
+    # option;
+    # for convenience, turn it into a `store_const` action
+    # (in Python 3.9+ one can use `argparse.BooleanOptionalAction`,
+    # but we want to maintain compatibility with Python 3.8)
+    action = parser_like.add_argument(*args, **kwargs)
+    long_flags = [arg for arg in args if arg.startswith('--')]
+    assert long_flags
+    if hide_complementary_options:
+        falsy_help_text = argparse.SUPPRESS
+    else:
+        falsy_help_text = 'Negate these flags: ' + ', '.join(args)
+    parser_like.add_argument(*('--no-' + flag[2:] for flag in long_flags),
+                             **{**kwargs,
+                                'const': False,
+                                'dest': action.dest,
+                                'help': falsy_help_text})
+    return action
+
+
+def get_cli_config(subtable, /, *args, **kwargs):
+    """
+    Get the `tool.line_profiler.<subtable>` configs and normalize
+    its keys (`some-key` -> `some_key`).
+
+    Arguments:
+        subtable (str):
+            Name of the subtable the CLI app should refer to (e.g.
+            'kernprof')
+        *args, **kwargs
+            Passed to `line_profiler.toml_config.get_config()`
+
+    Returns:
+        subconf_dict, path (tuple[dict, Path])
+            - `subconf_dict`: the combination of the
+              `tool.line_profiler.<subtable>` subtables of the
+              provided/looked-up config file (if any) and the default as
+              a dictionary
+            - `path`: absolute path to the config file whence the
+              config options are loaded
+    """
+    conf, source = get_config(*args, **kwargs)
+    cli_conf = {key.replace('-', '_'): value
```

Comment:
I've also tackled this in scriptconfig, although the code for doing so was complicated by internal changes to argparse in https://github.com/python/cpython/commit/c02b7ae4dd367444aa6822d5fb73b61e8f5a4ff9

Code is here:
https://gitlab.kitware.com/utils/scriptconfig/-/blob/main/scriptconfig/argparse_ext.py?ref_type=heads#L423

Might not be worth the extra complexity, but I figured I would point it out.